cmake/developer_package/compile_flags/os_flags.cmake

```diff
@@ -14,13 +14,13 @@ include(CheckCXXCompilerFlag)
 macro(ov_disable_deprecated_warnings)
     if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
         set(ov_c_cxx_deprecated "/wd4996")
-    elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
+    elseif(OV_COMPILER_IS_INTEL_LLVM)
```

Comment:
does LLVM Intel support MSVC-like options like it does for gcc on Linux?
If yes, we can unite paths for MSVC and Intel LLVM on Windows.

Reply:
[This](https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2024-2/invoke-the-compiler.html) link says that it does, but recommended path is MSVC style. Not sure what's the difference between those 2 from the functional/performance perspective